dotcom-rendering/scripts/dotenv/gen-dotenv.js

```diff
@@ -11,77 +11,99 @@ const checkEnv = async () => {
-			if (!env.match(regex)) valid = false;
+			const regex = new RegExp(`^${secret.key.replace('.', '\\.')}=.+`);
+			if (!env.match(regex)) return false;
 		}
 
-		return valid;
+		return true;
 	} catch (_err) {
 		return false;
 	}
 };
 
-const genEnv = async () => {
-	const env = process.env.NODE_ENV === 'production' ? 'prod' : 'dev';
-	const parameters = await getAwsSsmParameters(env);
-
+const genEnv = async (parameters) => {
 	let envString = '';
 	for (const secret of secrets) {
 		envString += `${secret.key}=${parameters[secret.key]}\n`;
 	}
 
 	await fs.writeFile(ENV_PATH, envString);
+
+	prompt('[scripts/dotenv] Generated .env file');
 };
 
-// eslint-disable-next-line @typescript-eslint/no-floating-promises
-(async () => {
+const prod = async () => {
 	try {
-		const validEnv = await checkEnv();
-		if (!validEnv) {
-			log(
-				'[scripts/dotenv] .env file is missing, attemting to generate it from AWS parameters...',
+		const params = await getAwsSsmParameters('prod');
+		await genEnv(params);
+		const isValid = await checkEnv();
+		if (!isValid) {
+			warn(
+				'[scripts/dotenv] Could not validate that the .env file was generated correctly',
+			);
+			throw new Error(
+				'Could not validate .env file was generated correctly',
 			);
-
-			await genEnv();
-
-			log('[scripts/dotenv] .env file written successfully');
-		} else {
-			log('[scripts/dotenv] valid .env file exists, moving on...');
 		}
 	} catch (err) {
-		if (err instanceof CredentialsProviderError) {
-			const PROD = process.env.NODE_ENV === 'production';
-			if (PROD) {
-				warn(
-					'[scripts/dotenv] could not generate .env file from AWS Parameter Store. Exiting',
-				);
-				throw err;
-			}
+		console.error(err);
+		process.exit(1);
+	}
+};
 
+const dev = async () => {
+	try {
+		const params = await getAwsSsmParameters('dev');
+		await genEnv(params);
+	} catch (err) {
+		if (
+			err instanceof CredentialsProviderError ||
+			// eslint-disable-next-line no-underscore-dangle -- '__type' is the only way to identify this exception type
+			err.__type === 'ExpiredTokenException'
+		) {
 			prompt(
 				'[scripts/dotenv] Could not load AWS credentials to generate .env file',
-				"[scripts/dotenv] This won't stop dotcom-rendering from working, it will just vary from PROD by:",
 			);
 
-			for (const secret of secrets) {
+			// We've failed to generate the .env file - but this is okay for the DEV environment
+			// If we already have an existing valid .env file, simply notify the user their .env was not updated.
+			// If we don't have a valid .env, notify them of the potential missing functionality.
+			const isValid = await checkEnv();
+			if (isValid) {
 				prompt(
-					`[scripts/dotenv]  * ${secret.key}: ${secret.missingMessage}`,
+					'[scripts/dotenv] .env file is valid but has not been updated',
+				);
+			} else {
+				prompt(
+					"[scripts/dotenv] This won't stop dotcom-rendering from working, it will just vary from PROD by:",
 				);
```

Comment:
This text seems to depend on another message for context but maybe it could have that context built into it?